### ISSUE TITLE
Gate Validation incorrectly rejects supported `SYC` gate 

### ISSUE THREAD

**Issue by senecameeks** [quantumlib/Cirq#6779]
**Description of the issue**

Gate validation improperly rejects `SYC` gate. When running internally and bypassing validation e.g `
remote_sampler = qe.CirqServerSampler(connector.grpc_client, "SYC01_10A_BLACK", "current", "iswap7_lower", use_json_serialization=True)` I'm able to get back a result. Likely a logical mismatch is occuring in `cirq-google/cirq_google/devices/grid_device.py` 

**How to reproduce the issue**
```
import time
import cirq
import cirq_google
from cirq_google.engine import get_engine
from cirq_google.engine.qcs_notebook import authenticate_user

authenticate_user()
engine = get_engine("cirq-tutorial")
engine.enable_streaming = False

id = 'SYC01_10A_BLACK'
processor = engine.get_processor(id)
sampler = processor.get_sampler(run_name="current", device_config_name="iswap7_lower")
duration = 600
elapsed_time = 0
# For "iswap7_lower", using cz, or sqrt_iswap.on, sqrt_iswap_like.on cause system errors
# for an iswap grid use SYC cirq_google.ops.SYC
# iswap_like = syc
qubit1 = cirq.GridQubit(8,4)
qubit2 = cirq.GridQubit(8,5)
circuit = cirq.Circuit(cirq_google.ops.SYC(qubit1, qubit2), cirq.measure(qubit1))


result = sampler.run_batch([circuit]*50)
print(result)
```
```
Outputs 
Job projects/cirq-tutorial/programs/prog-1DH5XNLJORTEK3Y4241029-101250/jobs/job-6HYWYUV7J2CH17UT241029-101250 on processor projects/cirq-tutorial/processors/SYC01_10A_BLACK failed. INVALID_PROGRAM: Operation cirq.FSimGate(theta=1.5707963705062866, phi=0.5235987901687622)(q(8, 4), q(8, 5)) contains a gate which is not supported.

```

**Comment by senecameeks** [comment]
I think the problem might be that `SYC` is not represented here in the default gatesets supported by Quantum Engine: https://github.com/qh-lab/pyle/blob/5640ffb2297f685239b3d6897f49028f3b82f058/src/pyle/quantum_engine/gatesets.py#L30

**Comment by BichengYing** [comment]
yes, you find the right spot! CC: @verult since he taught me about this file. I can create a simple PR to fix it

**Comment by BichengYing** [comment]
wait, there is no such serialization representation of SYC gate. https://github.com/quantumlib/Cirq/blob/2975d10912acd4183838f5c1ddba8f278faedb2a/cirq-google/cirq_google/api/v2/program.proto#L197-L220 So it is not that possible to quickly support it. @senecameeks Can you add serialization and deserialization logic for SYC gate?

**Comment by NoureldinYosri** [comment]
cirq-sync: SYC serialization is not implemented directly since it can be converted to an FSimGate

**Comment by eliottrosenberg** [comment]
@NoureldinYosri it also fails for the corresponding fSim gate. See [this](https://colab.sandbox.google.com/drive/1zY0pnCM_7LfgexsJWwMt_p5UKHxjknsp) colab.

**Comment by wcourtney** [comment]
It looks like the failure of FSim gate validation is because they apparently require the [`FSimViaModelTag`](https://github.com/quantumlib/Cirq/blob/ecf9ed750ca0d8fcc909b1ff382378ce98133b7b/cirq-google/cirq_google/ops/fsim_via_model_tag.py#L20). This was added by @BichengYing in https://github.com/quantumlib/Cirq/pull/6527.

I verified that including the tag passes validation, including when applied to the `SYC` gate ([colab](https://colab.sandbox.google.com/drive/1MFhH2XXjfjfPrtYTQYXOpbP6Fk6Nmqca)).

Example:
```
cirq_google.SYC(q1,q2).with_tags(cirq_google.FSimViaModelTag())
```

However, once the circuit is past validation and sent to the backends there are further issues that we should track on our internal repo.

@eliottrosenberg - is there additional work you'd like done on the client-side regarding validation here?

**Comment by eliottrosenberg** [comment]
@wcourtney with that tag, it would get translated to an arbitrary-angle fSim gate (i.e. a different gate internally), which is not what we want.

**Comment by pavoljuhas** [comment]
csynque meeting - check if this is still an issue after https://github.com/quantumlib/Cirq/pull/7093

**Comment by dstrain115** [comment]
@eliottrosenberg Can we verify if this is still an issue?
